Build failure: pianobar

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

```bash
sudo su - <<EOF
pushd /etc/nixos &&
  nix flake update &&
  nixos-rebuild --impure --flake . build;
EOF
```

... Above results in error similar to;

```
src/player.c:517:17: error: implicit declaration of function 'avcodec_close'; did you mean 'avio_close'? [-Wimplicit-fu>
  517 |                 avcodec_close (player->cctx);
      |                 ^~~~~~~~~~~~~
      |                 avio_close
make: *** [Makefile:110: src/player.o] Error 1
```


### Can Hydra reproduce this build failure?

Please select the Hydra Status.

### Link to Hydra build job

[Log of step 2 of build 313209385 of job...](https://hydra.nixos.org/build/313209385/nixlog/2)

### Relevant log output

```console
Running phase: unpackPhase
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/rfxpazrd134nfn7ha28nrj1c5j8m72k4-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: SHELL=/nix/store/qsydfxm1vq6q9jac2kq3r8kn0xdmsldf-bash-5.3p3/bin/bash PREFIX=\$\(out\)
    CC  src/main.c
src/main.c: In function 'BarMainGetLoginCredentials':
src/main.c:142:33: warning: ignoring return value of 'read' declared with attribute 'warn_unused_result' [-Wunused-resu>
  142 |                                 read (pipeFd[0], passBuf, sizeof (passBuf)-1);
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    CC  src/debug.c
    CC  src/player.c
src/player.c: In function 'finish':
src/player.c:517:17: error: implicit declaration of function 'avcodec_close'; did you mean 'avio_close'? [-Wimplicit-fu>
  517 |                 avcodec_close (player->cctx);
      |                 ^~~~~~~~~~~~~
      |                 avio_close
make: *** [Makefile:110: src/player.o] Error 1
```

### Additional context

<details><summary><code>/etc/nixos/flake.lock</code></summary>

> `/etc/nixos/flake.lock`

```nix
{
  "nodes": {
    "flake-parts": {
      "inputs": {
        "nixpkgs-lib": [
          "nur",
          "nixpkgs"
        ]
      },
      "locked": {
        "lastModified": 1733312601,
        "narHash": "sha256-4pDvzqnegAfRkPwO3wmwBhVi/Sye1mzps0zHWYnP88c=",
        "owner": "hercules-ci",
        "repo": "flake-parts",
        "rev": "205b12d8b7cd4802fbcb8e8ef6a0f1408781a4f9",
        "type": "github"
      },
      "original": {
        "owner": "hercules-ci",
        "repo": "flake-parts",
        "type": "github"
      }
    },
    "home-manager": {
      "inputs": {
        "nixpkgs": [
          "nixpkgs"
        ]
      },
      "locked": {
        "lastModified": 1763228015,
        "narHash": "sha256-1rYieMVUyZ3kK/cBIr8mOusxrOEJ1/+2MsOg0oJ7b3A=",
        "owner": "nix-community",
        "repo": "home-manager",
        "rev": "96156a9e86281c4bfc451236bc2ddfe4317e6f39",
        "type": "github"
      },
      "original": {
        "owner": "nix-community",
        "repo": "home-manager",
        "type": "github"
      }
    },
    "nixos-hardware": {
      "locked": {
        "lastModified": 1762847253,
        "narHash": "sha256-BWWnUUT01lPwCWUvS0p6Px5UOBFeXJ8jR+ZdLX8IbrU=",
        "owner": "NixOS",
        "repo": "nixos-hardware",
        "rev": "899dc449bc6428b9ee6b3b8f771ca2b0ef945ab9",
        "type": "github"
      },
      "original": {
        "owner": "NixOS",
        "ref": "master",
        "repo": "nixos-hardware",
        "type": "github"
      }
    },
    "nixpkgs": {
      "locked": {
        "lastModified": 1762977756,
        "narHash": "sha256-4PqRErxfe+2toFJFgcRKZ0UI9NSIOJa+7RXVtBhy4KE=",
        "owner": "nixos",
        "repo": "nixpkgs",
        "rev": "c5ae371f1a6a7fd27823bc500d9390b38c05fa55",
        "type": "github"
      },
      "original": {
        "owner": "nixos",
        "ref": "nixos-unstable",
        "repo": "nixpkgs",
        "type": "github"
      }
    },
    "nur": {
      "inputs": {
        "flake-parts": "flake-parts",
        "nixpkgs": [
          "nixpkgs"
        ]
      },
      "locked": {
        "lastModified": 1763246442,
        "narHash": "sha256-Fz67dJbhPhoISJqmqN3snBsvSeP+1NN9g3a6weRbFbM=",
        "owner": "nix-community",
        "repo": "NUR",
        "rev": "5ce75241e4aee42bc80c0323dcbd9b13d40d5169",
        "type": "github"
      },
      "original": {
        "owner": "nix-community",
        "repo": "NUR",
        "type": "github"
      }
    },
    "root": {
      "inputs": {
        "home-manager": "home-manager",
        "nixos-hardware": "nixos-hardware",
        "nixpkgs": "nixpkgs",
        "nur": "nur",
        "sops-nix": "sops-nix"
      }
    },
    "sops-nix": {
      "inputs": {
        "nixpkgs": [
          "nixpkgs"
        ]
      },
      "locked": {
        "lastModified": 1763069729,
        "narHash": "sha256-A91a+K0Q9wfdPLwL06e/kbHeAWSzPYy2EGdTDsyfb+s=",
        "owner": "Mic92",
        "repo": "sops-nix",
        "rev": "a2bcd1c25c1d29e22756ccae094032ab4ada2268",
        "type": "github"
      },
      "original": {
        "owner": "Mic92",
        "repo": "sops-nix",
        "type": "github"
      }
    }
  },
  "root": "root",
  "version": 7
}
```
</details>

<details><summary><code>/etc/nixos/home-manager/user/programs/pianobar.nix</code></summary>

> `home-manager/user/programs/pianobar.nix`


```nix
{
  config,
  ...
}:
{
  programs.pianobar = {
    enable = true;
    settings = {
      user = "redacted@example.com";
      password_command = "cat ${config.home.homeDirectory}/.config/sops-nix/secrets/pianobar/user";
    };
  };

  sops.secrets."pianobar/user" = {
    ## Note: path seems to be ignored and it instead places things according to
    ## how secrets are defined?
    path = "%r/pianobar/user";
  };
}
```
</details>


### System metadata

- system: `"x86_64-linux"`
- host os: `Linux 6.17.2, NixOS, 25.11 (Xantusia), 25.11.20251015.544961d`
- multi-user?: `yes`
- sandbox: `yes`
- version: `nix-env (Nix) 2.31.2`
- nixpkgs: `/nix/store/rcsmdj802sk3w9l8m0b0bvdy0jiw2wgr-source`

### Notify maintainers

`stdenv.mkDerivation` -> `meta.maintainers` is undefined.

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/pi/pianobar/package.nix
+++ b/pkgs/by-name/pi/pianobar/package.nix
@@ -32,9 +32,6 @@ stdenv.mkDerivation (finalAttrs: {
 
   makeFlags = [ "PREFIX=$(out)" ];
 
-  CC = "gcc";
-  CFLAGS = "-std=c99";
-
   meta = {
     changelog = "https://github.com/PromyLOPh/pianobar/raw/${finalAttrs.src.rev}/ChangeLog";
     description = "Console front-end for Pandora.com";

